src/optimizer/deliminator.cpp

```diff
@@ -294,4 +365,55 @@ bool Deliminator::RemoveInequalityJoinWithDelimGet(LogicalComparisonJoin &delim_
 	return found_all;
 }
 
+void Deliminator::TrySwitchSingleToLeft(LogicalComparisonJoin &delim_join) {
+	D_ASSERT(delim_join.join_type == JoinType::SINGLE);
+
+	// Collect RHS bindings
+	vector<ColumnBinding> join_bindings;
+	for (const auto &cond : delim_join.conditions) {
+		if (!IsEqualityJoinCondition(cond)) {
+			return;
+		}
+		if (cond.right->type != ExpressionType::BOUND_COLUMN_REF) {
```

Comment:
why do we only get the right conditions? Isn't it possible that we have a left delim join here and we should get the bindings from the left side?

Edit: It's because single joins inherently join right delim joins

Reply:
To be able to flip the join from SINGLE to LEFT, we need to know whether the RHS has, for some reason, been de-duplicated, e.g., by a random aggregate, that's all.

The semantics of the SINGLE join are to not join LHS values with more than one RHS value, but if the RHS is de-duplicated, this is guaranteed even with the LEFT join.